initializr/src/main/resources/static/js/start.js

```diff
@@ -58,4 +58,90 @@
         return versionA[3].localeCompare(versionB[3]);
     }
 
-}());
\ No newline at end of file
+}());
+
+$(function () {
+    var refreshDependencies = function (versionRange) {
+        var versions = new Versions();
+        $("#dependencies div.checkbox").each(function (idx, item) {
+            if ($(item).attr('data-range') === 'null' || versions.matchRange($(item).attr('data-range'))(versionRange)) {
+                $(item).show();
+            } else {
+                $(item).hide();
+                $("input", item).prop('checked', false);
+            }
+        });
+    };
+    var addTag = function (id, name) {
+        if ($("#starters div[data-id='" + id + "']").length == 0) {
+            $("#starters").append("<div class='tag' data-id='" + id + "'>" + name +
+                "<button type='button' class='close' aria-label='Close'><span aria-hidden='true'>&times;</span></button></div>");
+        }
+    };
+    var removeTag = function (id) {
+        $("#starters div[data-id='" + id + "']").remove();
+    }
+    refreshDependencies($("#bootVersion").val());
+    $("#type").on('change', function () {
+        $("#form").attr('action', $(this.options[this.selectedIndex]).attr('data-action'))
+    });
+    $("#artifactId").on('change', function () {
+        $("#baseDir").attr('value', this.value)
+    });
+    $("#bootVersion").on("change", function (e) {
+        refreshDependencies(this.value);
+    });
+
+    var starters = new Bloodhound({
+        datumTokenizer: Bloodhound.tokenizers.obj.whitespace('name', 'description'),
+        queryTokenizer: Bloodhound.tokenizers.whitespace,
+        identify: function (obj) {
+            return obj.id;
+        },
+        cache: false,
```

Comment:
I've disabled the autocompletion engine cache for now.
The autocompletion engine downloads the `dependencies.json` information and locally caches this data, if possible, in the browser's LocalStorage (by default, for a day).